### PR TITLE
pascal-p5: new port

### PR DESCRIPTION
**Pull request by kamischi**
#### Description
pascal-P5 is a Zuerich type ISO 7185 Pascal compiler.

###### Type(s)
- [x] enhancement

###### Tested on
macOS 10.14.6
Xcode 10.3

###### Verification 
Have you
- [*] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [*] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [*] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [*] checked your Portfile with `port lint`?
- [*] tried existing tests with `sudo port test`?
- [*] tried a full install with `sudo port -vst install`?
- [*] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/pascal-p5/Portfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+revision			0
+categories			lang
+platforms			darwin
+license				GPL-2
+maintainers			web.de:karl-michael.schindler
+
+description			P5, Zuerich type ISO 7185 Pascal compiler
+long_description	This is the 5th version of the Pascal-P compiler from Zuerich. \
+					It is modified to be ISO 7185 Pascal compliant, both in \
+					the implementation language, and in the language it processes.
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: tabs converted to spaces. Makes life easier when using different editors.

File: lang/pascal-p5/Portfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+homepage			https://www.standardpascal.com/p5.html
+
+fetch.type			git
+git.url				https://git.code.sf.net/p/pascalp5/code
+git.branch			75a0a8
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: unfortunately not.

File: lang/pascal-p5/Portfile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+build {	\
+	file mkdir ${worksrcpath}/build
+	system	-W ${worksrcpath} " \
+			fpc -Miso -ap -FEbuild source/pcom.pas && \
+			fpc -Miso -ap -FEbuild source/pint.pas "
+}
+
+destroot {
+	xinstall -m 755 -d						  ${destroot}${prefix}/bin/
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: see above

File: lang/pascal-p5/Portfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	xinstall -m 755 ${worksrcpath}/build/pint ${destroot}${prefix}/bin/
+	xinstall -m 755 ${worksrcpath}/build/pcom ${destroot}${prefix}/bin/
+	xinstall -m 755 ${filespath}/p5			  ${destroot}${prefix}/bin/
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: see above. should be fixed.

File: lang/pascal-p5/Portfile [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+installs_libs		no
+use_configure		no
+universal_variant	no
+
+build {	\
+	file mkdir ${worksrcpath}/build
+	system	-W ${worksrcpath} " \
+			fpc -Miso -ap -FEbuild source/pcom.pas && \
+			fpc -Miso -ap -FEbuild source/pint.pas "
+}
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: Actually, it comes with a Makefile, but it is setup for using the outdated gnu-pascal compiler gpc with a completely different set of options. IMHO patching the Makefile for fpc is less obvious than simply using the commands here.

File: lang/pascal-p5/Portfile [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+PortSystem			1.0
+
+name				pascal-p5
+version				1.4
+revision			0
+categories			lang
+platforms			darwin
+license				GPL-2
+maintainers			web.de:karl-michael.schindler
```
[the remark this answers is not in the document]
Comment (kamischi) [reply lang/pascal-p5/Portfile]: Fixed.

File: lang/pascal-p5/Portfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmetzger) [reply lang/pascal-p5/Portfile]: That's fine, in fact that's preferable, but then please use a consistent indentation depth.

File: lang/pascal-p5/Portfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (pmetzger) [reply lang/pascal-p5/Portfile]: Or you could provide your own Makefile in `files`... but either is fine.

File: python/py-distributed/Portfile [hunk 7]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-distributed
-version             2.3.0
+version             2.3.2
```
Comment (pmetzger) [inline python/py-distributed/Portfile]: This commit is unrelated and you need to get it out of your pull request.

File: python/py-jupyterlab/Portfile [hunk 8]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-jupyterlab
-version             1.1.0
+version             1.1.1
```
Comment (pmetzger) [inline python/py-jupyterlab/Portfile]: This commit is unrelated and you need to get it out of your pull request.

File: python/py-s3fs/Portfile [hunk 9]
```diff
@@ -5,7 +5,7 @@ PortGroup           python 1.0
 PortGroup           select 1.0
 
 name                py-s3fs
-version             0.3.3
+version             0.3.4
```
Comment (pmetzger) [inline python/py-s3fs/Portfile]: This commit is unrelated and you need to get it out of your pull request.

**Comment by pmetzger** [comment]
This still includes unintended files in the commit.

**Comment by kamischi** [comment]
I do not understand why they are still there. I reverted the corresponding commits and then rebase/stash. Also, the files do not contain the changes any longer. So, what should i do? Delete the files?

**Comment by pmetzger** [comment]
If I had direct access to your keyboard I would simply fix it. As it is, I suggest closing this PR and opening a new one with a fresh branch.

**Comment by cjones051073** [comment]
> If I had direct access to your keyboard I would simply fix it. As it is, I suggest closing this PR and opening a new one with a fresh branch.

Closing this PR is not really neccessary.

One way out is to copy out temporarily the changes you wish to save, then do a hard reset of the branch here w.r.t. primary master, then start again, i.e.

`git reset —hard origin/master`

Assuming origin is the primary macports repo in your checkout. If not, replace with what it is called.

Then, just make the changes again and finally force push here.

**Comment by kamischi** [comment]
> `git reset —hard origin/master`

I had more luck with `--hard`.

File: python/py-distributed/Portfile (hunk 7 above)
Comment (kamischi) [reply python/py-distributed/Portfile]: I hope that commit a8170a9 did the trick.

**Comment by pmetzger** [comment]
@cjones051073 What do you think of the current content? Should we merge?

**Comment by pmetzger** [comment]
Merged!
